### ISSUE TITLE
aws_elasticsearch_domain log_publishing_options log_type missing ES_APPLICATION_LOGS type

### ISSUE THREAD

**Issue by akosveres** [terraform-providers/terraform-provider-aws#5473]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.7
+ provider.aws v1.29.0

### Affected Resource(s)

* aws_elasticsearch_domain

### Terraform Configuration Files

```hcl
  log_publishing_options {
    log_type                 = "ES_APPLICATION_LOGS"
    cloudwatch_log_group_arn = "arn:aws:logs:eu-west-1:123456789123:log-group:/aws/aes/domains/es/application-logs"
    enabled                  = true
  }
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Expected Behavior

We should be able to setup `ES_APPLICATION_LOGS` type logging for elasticsearch. Once we setup logging in the AWS Console we see it as an option.

### Actual Behavior

We can only setup `INDEX_SLOW_LOGS` or `SEARCH_SLOW_LOGS`

```
Error: module.elasticsearch.aws_elasticsearch_domain.es: expected log_publishing_options.0.log_type to be one of [INDEX_SLOW_LOGS SEARCH_SLOW_LOGS], got ES_APPLICATION_LOGS
```

### Steps to Reproduce

Try and use `ES_APPLICATION_LOGS` type logging.
1. `terraform apply`

### References
The documentation states as well the two supported types of logging. https://www.terraform.io/docs/providers/aws/r/elasticsearch_domain.html#log_type

* #0000

**Comment by sgrodzicki** [comment]
This has already been reported in #5411 and fixed in #5474

**Comment by bflad** [comment]
Support for this has been merged in and will release with version 1.31.0 of the AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.31.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1310-august-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
